tests/test_image.py

```diff
@@ -1692,6 +1692,112 @@ def test_Image_binary_scalar_pow():
+    """
+    # Test that setting an Image initialized with an int array to a float array
+    # doesn't downcast and truncate values
+    farray = np.array([[1.2, 3.8], [5.6, 6.7]])
+    image = galsim.Image(np.zeros((2, 2), dtype=int))
+    image.array = farray
+    np.testing.assert_array_equal(image.array, farray)
+
+    for i in range(ntypes):
+        # First, check that
+        ref_array_view = ref_array.astype(types[i])
+        zeros_array = np.zeros_like(ref_array_view)
+
+        # Check that we can set an array appropriately;
+
+        # First, using the view of the underlying array
+        image1 = galsim.Image(zeros_array)
+        image1.array[:] = ref_array_view
+
+        np.testing.assert_array_equal(image1.array, ref_array_view)
+
+        # Second, using the setter
+        image2 = galsim.Image(zeros_array)
+        image2.array = ref_array_view
+
+        np.testing.assert_array_equal(image2.array, ref_array_view)
+
+        # Check that we do not allocate new memory when performing inplace
+        # operations on the Image array
+        internal_array = np.zeros_like(ref_array_view)
+        image3 = galsim.Image(internal_array)
+        image3_array_id = id(image3.array)
+        image3.array += ref_array_view
+
+        # Check that the id of the Image array attribute is unchanged
+        assert id(image3.array) == image3_array_id
+
+        # NumPy arrays will return views with new IDs, so we must check if the
+        # underlying data buffers are equal rather than just the IDs
+        assert image3.array.__array_interface__ == internal_array.__array_interface__
+
+        # Now we attempt verboten operations;
+
+        # Test that attempting to set an Image's array to an array of different
+        # shape raises an error
+        with assert_raises(ValueError):
+            image4 = galsim.Image(zeros_array)
+            image4.array = large_array
+
+        # Test that setting the array to a single values raises an error;
+        # note that this behavior can still be attained via, e.g.,
+        # ```
+        # image.array[:] = 1
+        # ```
+        with assert_raises(galsim.GalSimIncompatibleValuesError):
+            image5 = galsim.Image(zeros_array)
+            image5.array = 1
+
+
+@timer
+def test_Image_setter_alignment():
+    # Make an array that is _not_ memory aligned
+    # c.f. https://stackoverflow.com/a/9895986
+    m = n = 2
+    dtype = np.dtype(np.float64)
+    nbytes = m * n * dtype.itemsize
+    buf = np.empty(nbytes + 16, dtype=np.uint8)
+    start_index = -buf.ctypes.data % 16 + 1  # offset by 1 to unalign
+    unaligned_array = buf[start_index:start_index + nbytes].view(dtype).reshape(m, n)
+    unaligned_array[:] = 1
+    assert unaligned_array.ctypes.data % 16 != 0
+
+    # Make arrays that are memory-aligned by default
+    zeros_array = np.zeros((m, n))
+    ones_array = np.ones((m, n))
+    assert zeros_array.ctypes.data % 16 == 0
+    assert ones_array.ctypes.data % 16 == 0
```

Comment:
These will break if numpy is _not_ guaranteeing that arrays are memory-aligned by default, which I gather was not previously the case. We can get around this by writing a function to explicitly make memory-aligned arrays, but doing so may not be necessary